### PR TITLE
Add a test case to indicate shebang including invalid utf-8 char

### LINKED ISSUE #28: Get to 100% test coverage
https://codecov.io/gh/brettcannon/python-launcher

### PR DESCRIPTION
**Pull request by u5surf**
To fixes #28, cover the test for 
https://codecov.io/gh/brettcannon/python-launcher/src/master/src/cli.rs#L148

---
- [x] **UNDERSTOOD**: Brett is using this project to teach himself Rust, so I will
not feel offended if he chooses not to accept this PR in order for him to learn how to do
something on his own. (PRs to do something more idiomatically are very much appreciated,
though.)

### CODE REVIEW

**Comment by brettcannon** [comment]
@u5surf I just wanted to say thanks for the PR and that I am not purposefully ignoring it; just been too busy to work on this project lately. But I will eventually review it.

File: src/cli.rs [hunk 1]
```diff
@@ -322,4 +322,11 @@ mod tests {
     fn parse_python_shebang_tests(shebang: &str) -> Option<RequestedVersion> {
         parse_python_shebang(&mut shebang.as_bytes())
     }
+
+    #[test_case(&[0x23, 0x21, 0xc0, 0xaf] => None ; "invalid utf8")]
```
Comment (brettcannon) [inline src/cli.rs:326]:
```suggestion
    #[test_case(&[0x23, 0x21, 0xc0, 0xaf] => None ; "invalid UTF-8")]
```

**Comment by brettcannon** [comment]
Thanks, @u5surf !

**Comment by zachgates** [comment]
Hey there.

I was in the process of reading through the source code (trying to assimilate some Rust literacy), and I happened upon a typo in [`/src/cli.rs`][1]. I considered opening a new issue, but ultimately decided that the de facto label, `bug`, was not an apt descriptor. So, I dug a bit deeper — looking for someone to `git blame` — and wound up here.

<hr>

The aggregate changes contained in this PR were merged into [`main`][3] at 2361a7f from [`u5surf:issue-28`][4].

```rust
    #[test_case(&[0x23, 0x21, 0xc0, 0xaf] => None ; "invalid UTF-8")]
    fn parse_python_sheban_include_invalid_bytes_tests(
        mut shebang: &[u8],
    ) -> Option<RequestedVersion> {
        parse_python_shebang(&mut shebang)
    }
```

The word-perfect function name is `parse_python_shebang_include_invalid_bytes_tests` — as `sheban` becomes `shebang`. Extrapolating from your commit history, I presume that a supplementary commit (amending this functionally insignificant typo) would be the normative solution.

I would be happy to make the small contribution myself; however, naturally, you can do whatever you like.

<hr>

I would also like to extend my thanks to you for everything you have done here, across the board. Speaking as someone almost totally unfamiliar to Rust, I've gleaned a surprising amount of practical knowledge here — be it idiomatic or not, which I'm sure will increase with additional read-throughs. And, lastly: I'm glad there is finally this promising `py` port! I'm surprised we've gone this far down the road without one; it's only logical to have a cross-platform solution!

  [1]: ../blob/main/src/cli.rs
  [2]: ../blob/2361a7f/src/cli.rs#L326-L331
  [3]: ../tree/main
  [4]: https://github.com/u5surf/python-launcher/tree/issue-28

**Pull request merged**